docs/reference/api/metadata-stores/index.md

```diff
@@ -12,6 +12,7 @@ Metaxy has built-in support for the following metadata store types:
 ## Storage Only
 
 - [DeltaMetadataStore](./delta.md).
+- [LanceDBMetadataStore](./lancedb.md).
```

Comment:
LanceDB isn't a storage-only DB, right? similar to ducklake in this sense. 

We should put it into "databases" and explain that it can be used in both embedded and external compute variants.